### DISCUSSION TITLE
对话框服务的OnCloseAsync无法被正确触发

### DISCUSSION THREAD

**Discussion by gisakuya** [dotnetcore/BootstrapBlazor#3450] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

使用对话框服务DialogService打开一个窗口，然后在窗体里又使用DialogService打开另外一个窗体，依次关闭。
第一个窗体所注册的OnCloseAsyn无法被触发。使用的版本为：8.5.5

### Expected Behavior

每个注册到DialogService里的OnCloseAsyn都被正确调用

### Index.razor
```csharp
<Button OnClick="OpenDlg">点击打开弹窗</Button>

@code {
    async Task OpenDlg()
    {
        await DialogService.ShowCloseDialog<Test>("测试1", configureOption:opt=>
        {
            opt.OnCloseAsync = () =>
            {
                System.Console.WriteLine("关闭弹窗1");

                return Task.CompletedTask;
            };
        });
    }
}
```

### Test.razor
```csharp
<Button OnClick="OpenDlg">点击打开弹窗</Button>

@code {
    async Task OpenDlg()
    {
        await DialogService.ShowCloseDialog<Test>("测试2", configureOption: opt =>
        {
            opt.OnCloseAsync = () =>
            {
                System.Console.WriteLine("关闭弹窗2");

                return Task.CompletedTask;
            };
        });
    }
}
```

步骤：
依次点击按钮打开窗口，然后依次关闭，打印出的信息如下：
关闭弹窗2
关闭弹窗2

### .NET Version

8.0.101

**Comment by bb-auto** [comment]
@gisakuya Thank you for reporting. We will give triage later.

**Comment by ArgoZhang** [comment]
@gisakuya 请提供最小复现工程，不要让我新建工程，然后 cv 你的代码，帮我节约时间

**Comment by gisakuya** [comment]
> @gisakuya 请提供最小复现工程，不要让我新建工程，然后 cv 你的代码，帮我节约时间

链接: https://pan.baidu.com/s/1mvkCNaS_0QU7KWly11mqGg?pwd=6mf5 提取码: 6mf5 复制这段内容后打开百度网盘手机App，操作更方便哦

**Reply by ArgoZhang** [reply]
https://www.blazor.zone/dialog#MultipleDialog


现在国人一点都不替他人考虑问题，你问个问题，我就需要注册一个百度网盘，还要绑定一个手机号，开源太难了。建议遵循开源的正确做法，上传 zip 包或者提供仓库

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/22001478/a1b60987-8cc1-4e9d-ad01-259fda95d7b1)
